Assets/Scripts/Capabilities/BaseEnemyProjectileMotion.cs

```diff
@@ -21,4 +22,9 @@ void Update()
             Destroy(gameObject);
         }
     }
+
+    public void SetPossessionManager(PossessionManager sourcePossessionManager)
+    {
+        possessionManager = sourcePossessionManager;
+    }
```

Comment:
Why is this needed? Is this for the projectile? It seems weird to me that the projectile should need to know any information from the possession manager.

Reply:
This was part of the workaround for the hitbox logic. The projectile needs to know if the enemy that spawned it was possessed or not so it can determine what should be taking damage when it hits something

Reply:
Okay, this works then. Thanks for explaining.

As an aside, one way I've circumvented this issue in the past was by registering a GUID on the parent object, and set the same GUID on the projectile. Then, in the `OnTriggerEnter` (or equivalent) check, if the guids are equal, ignore the collision.

Reply:
I'm not sure if that would solve everything. It would prevent enemies from killing themselves with their own projectiles, but we would still need a way to ensure that possessed enemy projectiles damage other enemies. Possession is a lot more complicated than I thought it would be